app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,77 @@
+class SubmissionsController < ApplicationController
+  before_action :authenticate_user!, only: [:create, :create_comment]
+  before_action :authenticate_to_submit, only: :new
+
+  helper_method :posts, :post, :new_post, :new_comment
+
+  def new
+  end
+
+  def show
+  end
+
+  def index
+  end
+
+  def create
+    @post = Submission.new(post_params)
+    @post.user_id = current_user.id
+    if @post.save
+      flash[:alert] = "Your submission was saved"
+      redirect_to submission_path(@post.id)
+    else
+      flash[:alert] = @post.errors.messages
+      redirect_to new_submission_path
+    end
+  end
+
+  def create_comment
+    @comment = Submission.new(comment_params)
+    @comment.user_id = current_user.id
+    @comment.post_id = parent_post_id
+    if @comment.save
+      flash[:alert] = "Your submission was saved"
+      redirect_to submission_path(@comment.post_id)
+    else
+      flash[:alert] = @comment.errors.messages
+      redirect_to submission_path(parent_post_id)
+    end
+  end
+
+  private
+
+  def authenticate_to_submit
+    if !user_signed_in?
+      flash[:alert] = "You must be logged in to submit a post"
+      redirect_to new_user_session_path
+    end
+  end
+
+  def posts
+    @posts = Submission.where(post_id: nil)
```

Comment:
Use `||=` here to memoize the AR call.

Also, might want to consider adding a scope on Submission here if you find this query getting used over and over. Something like:

```ruby
class Submission < ApplicationRecord
  # ... other bits omitted ...

  # use `Submission.posts` to nab posts only
  scope :posts, ~> { where post_id: nil }

  # ... other bits omitted ...
end
```

you can find more info in the [docs](http://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope)